docs/reference/commandline/run.md

```diff
@@ -340,75 +340,43 @@ system's interfaces.
 $ docker run -e MYVAR1 --env MYVAR2=foo --env-file ./env.list ubuntu bash
 ```
 
-This sets simple (non-array) environmental variables in the container. For
-illustration all three
-flags are shown here. Where `-e`, `--env` take an environment variable and
-value, or if no `=` is provided, then that variable's current value, set via
-`export`, is passed through (i.e. `$MYVAR1` from the host is set to `$MYVAR1`
-in the container). When no `=` is provided and that variable is not defined
-in the client's environment then that variable will be removed from the
-container's list of environment variables. All three flags, `-e`, `--env` and
-`--env-file` can be repeated.
+The `-e`, `--env`, and `--env-file` flags allow you to set simple (non-array)
+environment variables in the container you're running.
 
-Regardless of the order of these three flags, the `--env-file` are processed
-first, and then `-e`, `--env` flags. This way, the `-e` or `--env` will
-override variables as needed.
+You can define the variable and its value when running the container:
 
 ```bash
-$ cat ./env.list
-TEST_FOO=BAR
-$ docker run --env TEST_FOO="This is a test" --env-file ./env.list busybox env | grep TEST_FOO
-TEST_FOO=This is a test
+$ docker run --env VAR1=value1 --env VAR2=value2 ubuntu env | grep VAR
+VAR1=value1
+VAR2=value2
 ```
 
-The `--env-file` flag takes a filename as an argument and expects each line
-to be in the `VAR=VAL` format, mimicking the argument passed to `--env`. Comment
-lines need only be prefixed with `#`
+You can also use variables that you've exported to your local environment:
```

Comment:
I included a small caveat after the snippet. I agree that you should make the declaration of variables as explicit as possible, and I bet most uses for setting variables from the environment where for situations that are now covered by secrets.